lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,52 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(account, config, request) {
+
+  var confirmLogin = config.signinConfirmation && config.signinConfirmation.enabled
+  if (confirmLogin) {
+
+    var uid = account.uid.toString('hex')
+    var email = account.email
+
+    // If feature enabled, always enable for these emails
+    var emailDomain = account.email.substring(email.indexOf('@'), email.length).toLocaleLowerCase()
+    var isValidEmail = config.signinConfirmation.allowEmails.indexOf(emailDomain) > -1
+    if (isValidEmail) {
+      return true
+    }
```

Comment:
I think we need to move this check down after the `isValidContext` check, otherwise @mozilla.com users logging in on their Android device are going to have a bad time.